### ISSUE TITLE
Compilation errors ignored during processing

### ISSUE THREAD

**Issue by cordahi** [bsorrentino/maven-annotation-plugin#81]
The compiler invoked by the plugin implies that there were errors, however the plugin seems to ignore them and permits the maven compilation to continue.

```
4 errors
1 warning
```

As far as I can tell, the `AnnotationProcessorCompiler` discards the [CompilerResult](https://github.com/bsorrentino/maven-annotation-plugin/blob/v3.3.3/src/main/java/org/bsc/maven/plugin/processor/AnnotationProcessorCompiler.java#L389). The `execute()` method should return [result.isSuccess()](https://github.com/sonatype/plexus-compiler/blob/master/plexus-compiler-api/src/main/java/org/codehaus/plexus/compiler/CompilerResult.java#L57) required by the `CompilationTask.call()` method which instead unconditionally assumes [success](https://github.com/bsorrentino/maven-annotation-plugin/blob/v3.3.3/src/main/java/org/bsc/maven/plugin/processor/AnnotationProcessorCompiler.java#L419).

**Comment by cordahi** [comment]
The following diff fixes the error.  I applied it to v3.3.3, but it can also be applied to develop head

```
--- a/src/main/java/org/bsc/maven/plugin/processor/AnnotationProcessorCompiler.java
+++ b/src/main/java/org/bsc/maven/plugin/processor/AnnotationProcessorCompiler.java
@@ -296,7 +296,7 @@ public class AnnotationProcessorCompiler implements JavaCompiler {
     }
     
     
-    private void execute(   final Iterable<String> options, 
+    private Boolean execute(final Iterable<String> options, 
                             final Iterable<? extends JavaFileObject> compilationUnits,
                             final Writer w ) throws Exception 
     {
@@ -389,6 +389,8 @@ public class AnnotationProcessorCompiler implements JavaCompiler {
         result = PlexusJavaCompilerWithOutput.INSTANCE.performCompile(javacConf);
         
         out.println( result.toString() ); out.flush();
+
+        return result.isSuccess();
     }
     
     @Override
@@ -415,8 +417,7 @@ public class AnnotationProcessorCompiler implements JavaCompiler {
             @Override
             public Boolean call() {
                 try {
-                    execute(options, compilationUnits, out);
-                    return true;
+                    return execute(options, compilationUnits, out);
                 } catch (final Exception ex) {
                     diagnosticListener.report( new Diagnostic<JavaFileObject>() {
                         @Override
```

Now the plugin correctly detects and fails the build
```
4 errors
1 warning

[ERROR] error on execute: use -X to have details 
```

**Comment by bsorrentino** [comment]
Hi @cordahi 

Thanks for deep analysis of the problem, I'll apply your suggestion asap

**Comment by bsorrentino** [comment]
fix released in `4.0`

**Comment by cordahi** [comment]
Thanks @bsorrentino, I should have mentioned that one of the reasons I was using v3.3.3 was due to issue #82
